src/builder.rs

```diff
@@ -547,20 +675,150 @@ impl Builder {
             self.anchor,
             self.spends,
             self.outputs,
+            extsks,
         )
     }
+
+    /// Builds a bundle containing the given spent notes and outputs along with their
+    /// metadata, for inclusion in a PCZT.
+    pub fn build_for_pczt(
+        self,
+        rng: impl RngCore,
+    ) -> Result<(crate::pczt::Bundle, SaplingMetadata), Error> {
+        match self.zip212_enforcement {
+            Zip212Enforcement::Off | Zip212Enforcement::GracePeriod => {
```

Comment:
Why fail in the grace period? (Non-blocking; only matters in practice for tests.)